canonical-web-and-design/maas-ui
# refactor(ui): refactor machine test details to work with controllers

## Done

- Refactored machine test details components to work with controllers

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the React commissioning tab for a controller
- Open the action dropdown and click "View details..." for a test and check that you get taken to the details page
- Refresh the page and check that everything renders as before
- Click "Back to test results" and check you get taken back to the commissioning test list
- Check that the machine commissioning and tests tab work as before

## Fixes

Fixes canonical-web-and-design/app-tribe#943

## Screenshots

![Screenshot 2022-06-08 at 16-25-05 bolla commissioning bolla MAAS](https://user-images.githubusercontent.com/25733845/172546493-8f1c060c-8b4c-4eb6-beec-d679791cb211.png)
